packages/editor/src/components/document-outline/index.js

```diff
@@ -76,7 +76,11 @@ export const DocumentOutline = ( {
 	let prevHeadingLevel = 1;
 
 	// Not great but it's the simplest way to locate the title right now.
-	const titleNode = document.querySelector( '.editor-post-title__input' );
+	const titleNode =
+		document.querySelector( '.editor-post-title__input' ) ||
+		window.frames[ 'editor-canvas' ]?.document.querySelector(
```

Comment:
I really don't like this. :) Could we call `onSelect` for the `PostTitle`, which is in this package? I guess we'd have to move the local state to the package store first.

Reply:
What do you mean by `state`? And which part exactly you don't like? I mean querying for nodes is bad, but I don't think I deviated here from the existing `document.querySelector( '.editor-post-title__input' )`. 😄

Reply:
My understanding is that state is the [isSelected](https://github.com/WordPress/gutenberg/blob/0b5ca2239f012e676d70d5011f9f96c503cc8220/packages/editor/src/components/post-title/index.js#L44) that the `PostTitle` component has. This would mean that the code would be something like the following? 🤔

```javascript
const { selectPostTitle } = useDispatch( editorStore );

const isPostTypeSelected = useSelect(
	( select ) => {
		return select( editorStore ).isPostTitleSelected();
	},
	[]
);
```

If so, I think the problem with #48189 may be resolved.